adafruit_progressbar/__init__.py

```diff
@@ -0,0 +1,583 @@
+        :type color: int/None/Tuple[int, int, int]
+        """
+        self._fill_color = color
+        if color is None:
+            self._palette[0] = 0x00
+            self._palette.make_transparent(0)
+        else:
+            self._palette[0] = color
+            self._palette.make_opaque(0)
+
+    @property
+    def bar_color(self) -> int:
+        """The color of the bar's fill
+
+        :rtype: int/None
+        """
+
+        return self._bar_color
+
+    @bar_color.setter
+    def bar_color(self, color: Union[int, Tuple[int, int, int]]) -> None:
+        """Sets the color of the bar
+
+        :param color: The color to use for the bar
+        :type color: int/None/Tuple[int, int, int]
+
+        :rtype: None
+        """
+
+        self._bar_color = color
+
+        if color is None:
+            self._palette[2] = 0x00
+            self._palette.make_transparent(2)
+        else:
+            self._palette[2] = color
+            self._palette.make_opaque(2)
+
+    @property
+    def value(self) -> Union[int, float]:
+        """
+        The current value of the control, used to determine its progress/ratio
+        :rtype: int/float
+        """
+        return self._value
+
+    @value.setter
+    def value(self, value: Union[int, float]) -> None:
+        """Sets the current value of the progress within the min-max range
+
+        :param value: The new value for the progress status
+        :type value: int/float
+
+        :rtype: None
+        """
+
+        assert isinstance(
+            value, (int, float)
+        ), "The value to set must be either an integer or a float"
+
+        assert (
+            self.minimum <= value <= self.maximum
+        ), f"The value must be between minimum ({self.minimum}) and maximum ({self.maximum})"
+
+        # Save off the previous value, so we can pass it in the
+        # call to "Render"
+        self._old_value = self._value
+        self._value = value
+        # Convert value to float since we may be dealing with
+        # integer types, and we can't work with integer division
+        # to get a ratio (position) of "value" within range.
+        self._set_progress(self.get_value_ratio(value))
+
+    @property
+    def progress(self) -> float:
+        """Gets the current displayed value of the widget.
+
+        :return: The current progress ratio
+        :rtype: float
+        """
+        return self._progress
+
+    @progress.setter
+    def progress(self, value: float) -> None:
+        """Sets the current displayed value of the widget.
+
+        :param value: The new value which should be displayed by the progress
+                            bar. Must be between 0.0-1.0
+        :type value: float
+
+        :rtype: None
+        """
+
+        assert [isinstance(value, (float, int)), "'progress' must be an int or a float"]
+
+        assert 0.0 <= value <= 100.0, "'progress' must be between 0 and 100"
+
+        self._set_progress(value)
```

Comment:
If I understood correctly we were going to include a statement like this: 
```
self.value = self.minimum + ((self.maximum - self.minimum) * value)
```
here inside the progress setter. As well as leave a note in the comment letting the user know that `progress` could approximate values and that `value` should be used if it needs to be more precise.

I tested this out and it seems to be working correctly across many different min and max values include negatives.